{"name": "Проверка на ввод почты и переход после клика по кнопке \"Восстановить\"", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16", "trace": "self = <tests.test_recovery_pass.TestRecoveryPassword object at 0x0000029846677650>\nbase_page = <web_pages.base_page.BasePage object at 0x00000298466DA150>, main_page = <web_pages.main_page.MainPage object at 0x00000298466DA180>\nrecovery_password_page = <web_pages.recovery_password_page.PasswordRecoveryPage object at 0x00000298466DA060>\n\n    @allure.title('Проверка на ввод почты и переход после клика по кнопке \"Восстановить\"')\n    def test_enter_email_and_click_reset(self, base_page, main_page, recovery_password_page):\n        base_page.open_link(Urls.url_restore)\n        recovery_password_page.set_email_for_reset_password(PersonData.user_login)\n        recovery_password_page.click_reset_button()\n>       recovery_password_page.find_save_button()\n\ntests\\test_recovery_pass.py:22: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\nweb_pages\\recovery_password_page.py:25: in find_save_button\n    self.wait_until_element_visibility(AuthForgotPasswordlocators.SAVE_BUTTON)\nweb_pages\\base_page.py:42: in wait_until_element_visibility\n    return WebDriverWait(self.driver, 10).until(EC.visibility_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"18c68224-721b-491e-93fd-8d1eb16aa152\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x000002984666EFC0>, message = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nE       WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nE       NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\nE       dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n\nvenv\\Lib\\site-packages\\selenium\\webdriver\\support\\wait.py:105: TimeoutException"}, "steps": [{"name": "Открываем ссылку", "status": "passed", "parameters": [{"name": "url", "value": "'https://stellarburgers.nomoreparties.site/forgot-password'"}], "start": 1721491447118, "stop": 1721491447723}, {"name": "Вводим емейл в поле для восстановления пароля", "status": "passed", "steps": [{"name": "Вставить текст 'Nuykin_9@yandex.ru'", "status": "passed", "steps": [{"name": "Дождаться кликабельности элемента", "status": "passed", "parameters": [{"name": "locator", "value": "('xpath', '//label[text()=\"Email\"]/following-sibling::input')"}], "start": 1721491447723, "stop": 1721491447765}], "parameters": [{"name": "locator", "value": "('xpath', '//label[text()=\"Email\"]/following-sibling::input')"}, {"name": "text", "value": "'Nuykin_9@yandex.ru'"}], "start": 1721491447723, "stop": 1721491447840}], "parameters": [{"name": "email", "value": "'Nuykin_9@yandex.ru'"}], "start": 1721491447723, "stop": 1721491447840}, {"name": "Нажимаем на кнопку Восстановить", "status": "passed", "steps": [{"name": "Переместиться до элемента и кликнуть", "status": "passed", "parameters": [{"name": "locator", "value": "('xpath', '//button[text()=\"Восстановить\"]')"}], "start": 1721491447840, "stop": 1721491448130}], "start": 1721491447840, "stop": 1721491448130}, {"name": "Найти кнопку Сохранить", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n\n", "trace": "  File \"C:\\Users\\User\\Diplom\\Diplom_3\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"C:\\Users\\User\\Diplom\\Diplom_3\\web_pages\\recovery_password_page.py\", line 25, in find_save_button\n    self.wait_until_element_visibility(AuthForgotPasswordlocators.SAVE_BUTTON)\n  File \"C:\\Users\\User\\Diplom\\Diplom_3\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"C:\\Users\\User\\Diplom\\Diplom_3\\web_pages\\base_page.py\", line 42, in wait_until_element_visibility\n    return WebDriverWait(self.driver, 10).until(EC.visibility_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\User\\Diplom\\Diplom_3\\venv\\Lib\\site-packages\\selenium\\webdriver\\support\\wait.py\", line 105, in until\n    raise TimeoutException(message, screen, stacktrace)\n"}, "steps": [{"name": "Дождаться видимости элемента", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n\n", "trace": "  File \"C:\\Users\\User\\Diplom\\Diplom_3\\venv\\Lib\\site-packages\\allure_commons\\_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"C:\\Users\\User\\Diplom\\Diplom_3\\web_pages\\base_page.py\", line 42, in wait_until_element_visibility\n    return WebDriverWait(self.driver, 10).until(EC.visibility_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\User\\Diplom\\Diplom_3\\venv\\Lib\\site-packages\\selenium\\webdriver\\support\\wait.py\", line 105, in until\n    raise TimeoutException(message, screen, stacktrace)\n"}, "parameters": [{"name": "locator", "value": "('xpath', '//button[text()=\"Сохранить\"]')"}], "start": 1721491448130, "stop": 1721491458266}], "start": 1721491448130, "stop": 1721491458267}], "parameters": [{"name": "driver_do", "value": "'firefox'"}], "start": 1721491447117, "stop": 1721491458268, "uuid": "f510cc1c-856e-437a-a4e0-e3e6f0c017a7", "historyId": "fc803ce619433454a2808771cb76e189", "testCaseId": "3be43cfb5f0fec4070e5cae7f81a3b15", "fullName": "tests.test_recovery_pass.TestRecoveryPassword#test_enter_email_and_click_reset", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_recovery_pass"}, {"name": "subSuite", "value": "TestRecoveryPassword"}, {"name": "host", "value": "DESKTOP-UBJ37DP"}, {"name": "thread", "value": "21696-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_recovery_pass"}]}